src/mlpack/core/data/csv_parser_impl.hpp

```diff
@@ -0,0 +1,203 @@
+ * This can be used to load data into any matrix, i.e. arma and bandicoot
+ * in future.
+ *
+ * https://gitlab.com/conradsnicta/armadillo-code/-/blob/10.5.x/include/armadillo_bits/diskio_meat.hpp
+ * Copyright 2008-2016 Conrad Sanderson (http://conradsanderson.id.au)
+ * Copyright 2008-2016 National ICT Australia (NICTA)
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_CSV_PARSER_IMPL_HPP
+#define MLPACK_CORE_DATA_CSV_PARSER_IMPL_HPP
+
+#include "csv_parser.hpp"
+
+namespace mlpack
+{
+namespace data
+{
+  /**
+   * Given the address of a martix element(val)
+   * sets it equal to the provided value(token)
+   * example calling: convert_token<eT>(x.at(row, col), token)
+   */
+  template<typename MatType>
+  bool ConvertToken(typename MatType::elem_type& val, const std::string& token)
+  {
+    const size_t N = size_t(token.length());
+
+    if (N == 0)
+    {
+      val = typename MatType::elem_type(0);
+      return true;
+    }
+
+    const char* str = token.c_str();
+
+    if ((N == 3) || (N == 4))
+    {
+      const bool neg = (str[0] == '-');
+      const bool pos = (str[0] == '+');
+
+      const size_t offset = ((neg || pos) && (N == 4)) ? 1 : 0;
+
+      const char sig_a = str[offset];
+      const char sig_b = str[offset+1];
+      const char sig_c = str[offset+2];
+
+      if (((sig_a == 'i') || (sig_a == 'I')) &&
+          ((sig_b == 'n') || (sig_b == 'N')) &&
+	  ((sig_c == 'f') || (sig_c == 'F')))
+      {
+        val = neg ? -(std::numeric_limits<typename MatType::elem_type>::infinity()) :
+	            std::numeric_limits<typename MatType::elem_type>::infinity();
+        return true;
+      }
+      else if (((sig_a == 'n') || (sig_a == 'N')) &&
+	       ((sig_b == 'a') || (sig_b == 'A')) &&
+	       ((sig_c == 'n') || (sig_c == 'N')))
+      {
+        val = std::numeric_limits<typename MatType::elem_type>::quiet_NaN();
+        return true;
+      }
+    }
+
+    char* endptr = nullptr;
+
+    if ((std::is_floating_point<typename MatType::elem_type>::value) ||
+        (std::is_integral<typename MatType::elem_type>::value))
```

Comment:
Not sure if `std::is_integral` make sense here.  Only `std::is_floating_point` should be used here, as
[std::strtod()](https://en.cppreference.com/w/cpp/string/byte/strtof) (as used in the code below) is for floating point values, not integers.

Reply:
Oh, okay, I will make the change.